src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs

```diff
@@ -92,7 +94,8 @@ protected virtual void SerializeToFile(OmniSharpProjectSnapshot projectSnapshot,
             // by the time we move the tempfile into its place
             using (var writer = tempFileInfo.CreateText())
             {
-                _serializer.Serialize(writer, projectSnapshot);
+                var projectRazorJson = new ProjectRazorJson(publishFilePath, projectSnapshot.InternalProjectSnapshot);
```

Comment:
This is the primary change. We need access to the internal project snapshot, hence the other accessibility changes in the PR.